src/common/src/catalog/mod.rs

```diff
@@ -136,7 +136,7 @@ pub fn cdc_table_name_column_desc() -> ColumnDesc {
 /// The local system catalog reader in the frontend node.
 #[async_trait]
 pub trait SysCatalogReader: Sync + Send + 'static {
-    async fn read_table(&self, table_id: &TableId) -> Result<Vec<OwnedRow>>;
+    async fn read_table(&self, table_id: &TableId) -> Result<Vec<OwnedRow>, BoxedError>;
```

Comment:
Will `BoxedError` be abused after removing `RwError`?

Reply:
Since this is a trait, we have no information about how the users may have implemented it, so does the error type. It seems a common practice to let trait methods return `Box<dyn Error>`. Similar ideas are applied to `sqllogictest::AsyncDB`.